external-crates/move/crates/move-analyzer/src/symbols.rs

```diff
@@ -2173,9 +2404,36 @@ impl<'a> ParsingSymbolicator<'a> {
             | E::Unit
             | E::Continue(_)
             | E::Spec(_)
-            | E::Match(_, _) // TODO support it
-            | E::UnresolvedError
-            => (),
+            | E::UnresolvedError => (),
+        }
+    }
+
+    fn match_pattern_symbols(&mut self, sp!(_, pattern): &P::MatchPattern) {
+        use P::MatchPattern_ as MP;
+        match pattern {
+            MP::PositionalConstructor(chain, sp!(_, v)) => {
+                self.chain_symbols(chain);
+                v.iter().for_each(|e| {
+                    if let P::Ellipsis::Binder(m) = e {
+                        self.match_pattern_symbols(m);
+                    }
+                })
+            }
+            MP::FieldConstructor(chain, sp!(_, v)) => {
+                self.chain_symbols(chain);
+                v.iter().for_each(|e| {
+                    if let P::Ellipsis::Binder((_, m)) = e {
+                        self.match_pattern_symbols(m);
+                    }
+                })
+            }
+            MP::Name(_, chain) => self.chain_symbols(chain),
+            MP::Or(m1, m2) => {
+                self.match_pattern_symbols(m1);
+                self.match_pattern_symbols(m2);
```

Comment:
Small nit: we may prefer to process `m2` first so the def location is the soonest one possible.